lib/Song.rb

```diff
@@ -0,0 +1,74 @@
+require 'pry'
+
+class Song
+
+  attr_accessor :name, :artist, :genre
+
+  @@all = []
+
+  def initialize(name, artist=nil, genre=nil)
+    @name = name
+    self.artist = artist if artist
+    self.genre=genre if genre
+    @genre = genre
```

Comment:
This is already happening in your `genre=` method, so it doesn't need to happen here.

Reply:
Updated.